compiler/rustc_middle/src/ty/mod.rs

```diff
@@ -195,6 +195,8 @@ pub struct AssocItem {
     pub container: AssocItemContainer,
 }
 
+impl Eq for AssocItem {}
```

Comment:
Are you sure you can implement Eq when one of the fields doesn't implement it? I think that might overwrite some fields if you store it in a hashmap, instead of having separate entries.